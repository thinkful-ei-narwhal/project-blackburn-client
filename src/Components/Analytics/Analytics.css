.graphs-container {
    margin: 20px;
    padding: 10px;
    box-shadow: 2px 2px 2px 2px lightgray;
    background-color: white;
}


.averages {
    margin: 20px;
    background-color: white;
    box-shadow: 2px 2px 2px 2px lightgray;
    padding: 20px;
}

.analytics-container {
    margin-top: 2vh;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
}

.averages-container {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    align-items: flex-start;
}

@media screen and (min-width: 400px) {
    .analytics-container {
        margin-top: 2vh;
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: flex-start;
    }
}


@media screen and (min-width: 1000px) {
    .analytics-container {
        margin-top: 2vh;
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: flex-start;
    }
}