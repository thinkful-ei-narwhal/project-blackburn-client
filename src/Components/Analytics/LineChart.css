.score-graph {
    margin: 10px;
    box-shadow: 5px 5px 5px 5px lightgray;
    padding: 20px;
}

.score-graph:hover {
    box-shadow: 10px 10px 10px 10px lightgray;

}

.wpm-graph {
    margin: 10px;
    box-shadow: 5px 5px 5px 5px lightgray;
    padding: 20px;
}

.wpm-graph:hover {
    box-shadow: 10px 10px 10px 10px lightgray;
}

