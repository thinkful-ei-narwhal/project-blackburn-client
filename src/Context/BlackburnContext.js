import React, { Component } from 'react';
import { render } from '@testing-library/react';
import TokenService from '../Services/token-service';
import ApiService from '../Services/auth-api-service';

const BlackBurnContext = React.createContext({
  user: {},
  error: null,
  setError: () => {},
  clearError: () => {},
  setUser: () => {},
  processLogin: () => {},
});

export default BlackBurnContext;

export class BlackburnProvider extends Component {
  constructor(props) {
    super(props);
    const state = {
      user: {},
      error: null,
    };
    const payload = TokenService.parseAuthToken();
    if (payload)
      state.user = {
        id: payload.user_id,
        username: payload.sub,
        avatar: payload.avatar,
      };

    this.state = state;
  }

  setError = (error) => {
    console.error(error);
    this.setState({ error });
  };

  clearError = () => {
    this.setState({ error: null });
  };

  setUser = (user) => {
    this.setState({ user });
  };

  processLogin = (token) => {
    TokenService.saveAuthToken(token);
    const payload = TokenService.parseAuthToken();
    this.setUser({
      id: payload.user_id,
      username: payload.sub,
      avatar: payload.avatar,
    });
  };

  render() {
    const value = {
      user: this.state.user,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setUser: this.setUser,
      processLogin: this.processLogin,
    };
    return (
      <BlackBurnContext.Provider value={value}>
        {this.props.children}
      </BlackBurnContext.Provider>
    );
  }
}
