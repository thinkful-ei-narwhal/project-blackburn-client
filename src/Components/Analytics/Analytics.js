import React from 'react';
import LineChart from './LineChart';
import { Container, Row, Col } from 'react-grid-system';
import BlackBurnContext from '../../Context/BlackburnContext';
import './Analytics.css';
export default class Analytics extends React.Component {
  static contextType = BlackBurnContext;

  componentDidMount() {
    this.context.getMyScores();
    this.context.getTopTenScores();
  }

  render() {
    console.log(this.context.myScores);
    let arrWPM = this.context.myScores.map((data) => {
      return data.wpm;
    });
    let avgWPM = arrWPM.reduce((a, b) => a + b, 0) / arrWPM.length;
    let roundedWPM = avgWPM.toFixed(2);
    let scoreArr = this.context.myScores.map((data) => {
      return data.score;
    });
    let avgScore = scoreArr.reduce((a, b) => a + b, 0) / scoreArr.length;
    let roundedScore = avgScore.toFixed(2);
    return (
      <Container>
        <Row>
          <Col sm={6}>
            <LineChart myScores={this.context.myScores} />
          </Col>
          <Col sm={6}>
            <div className="averages">
              <div>
                {' '}
                <h3>Average Words Per Minute </h3> {roundedWPM}
              </div>
              <div>
                {' '}
                <h3>Average Score </h3> {roundedScore}
              </div>
            </div>
          </Col>
        </Row>
      </Container>
    );
  }
}
