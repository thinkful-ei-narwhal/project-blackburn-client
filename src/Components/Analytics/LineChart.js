import React from 'react';
import { VictoryChart, VictoryTheme, VictoryLine } from 'victory';
import BlackBurnContext from '../../Context/BlackburnContext';
import './LineChart.css';
import scoreboardService from '../../Services/scoreboard-api-service';
import { ContinuousColorLegend } from 'react-vis';

export default class LineChart extends React.Component {
  static contextType = BlackBurnContext;
  state = {
    sortedScores: [],
    sortedWPM: [],
    error: '',
  };
  componentDidMount() {
    scoreboardService
      .getSortedScores(this.context.user.id, 'sortdate')
      .then((res) =>
        this.setState({ sortedScores: res.score, sortedWPM: res.wpm })
      )
      .catch((err) => this.setState({ error: err.error }));
  }

  render() {
    console.log(this.state.sortedScores);
    const formatScoreDate = this.state.sortedScores.map((data) => {
      const options = { year: 'numeric', month: 'long', day: 'numeric' };
      const newDate = new Date(data.date_trunc).toLocaleDateString(
        'en-US',
        options
      );
      return { date_trunc: newDate, max: data.max };
    });
    const formatWPMDate = this.state.sortedWPM.map((data) => {
      const options = { year: 'numeric', month: 'long', day: 'numeric' };
      const newDate = new Date(data.date_trunc).toLocaleDateString(
        'en-US',
        options
      );
      return { date_trunc: newDate, max: data.max };
    });
    const wpmData = formatWPMDate.map((data) => {
      return { x: data.date_trunc, y: data.max };
    });
    const scoreData = formatScoreDate.map((data) => {
      return { x: data.date_trunc, y: data.max };
    });
    console.log(formatScoreDate);
    return (
      <div className="graphs">
        {this.state.error && <div className="error">{this.state.error}</div>}
        <div className="score-graph">
          <h3> Score Over Time </h3>
          <VictoryChart domainPadding={20} theme={VictoryTheme.material}>
            <VictoryLine
              style={{
                data: { stroke: '#c43a31' },
                parent: { border: '1px solid #ccc' },
              }}
              data={scoreData}
            />
          </VictoryChart>
        </div>
        <div className="wpm-graph">
          <h3> WPM Over Time </h3>
          <VictoryChart domainPadding={20} theme={VictoryTheme.material}>
            <VictoryLine
              style={{
                data: { stroke: '#c43a31' },
                parent: { border: '1px solid #ccc' },
              }}
              data={wpmData}
            />
          </VictoryChart>
        </div>
      </div>
    );
  }
}
