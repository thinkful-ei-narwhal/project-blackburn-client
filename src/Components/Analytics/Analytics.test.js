import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ReactDOM from 'react-dom'
import  BlackburnContext  from '../../Context/BlackburnContext'
import Analytics from './Analytics';
import BlackBurnContext from '../../Context/BlackburnContext';

describe(`Adds header to dom without crashing`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(
        <BlackBurnContext.Provider
            value = {{}}
        >
            <Analytics />
        </BlackBurnContext.Provider>
    )
    .find('.analytics-container')
    expect(toJson(wrapper)).toMatchSnapshot()
    })
})