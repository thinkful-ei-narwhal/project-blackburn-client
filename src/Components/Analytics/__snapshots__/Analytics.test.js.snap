// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds header to dom without crashing renders without crashing 1`] = `null`;
